modules/standard/Path.chpl

```diff
@@ -286,8 +284,27 @@ proc file.realPath(): string throws {
         }
         else 
           return false;
-     }
-     else
-       compilerError("Target platform should have Unix like environment");
-  } 
+    } else {
+      compilerError("Target platform should have Unix like environment");
+    }
+  }
+
+  /* Helper to determine if we are in a Unix environment.  Not sure yet how
+     useful this will be to other functions, but I didn't want to write that
+     big conditional in the function itself */
+  private proc isUnixEnv() param {
+    if (CHPL_TARGET_PLATFORM == "linux64" ||
+        CHPL_TARGET_PLATFORM == "linux32" ||
+        CHPL_TARGET_PLATFORM == "darwin" ||
+        CHPL_TARGET_PLATFORM == "netbsd64" ||
+        CHPL_TARGET_PLATFORM == "netbsd32" ||
+        CHPL_TARGET_PLATFORM == "cray-xk" ||
+        CHPL_TARGET_PLATFORM == "cray-xe" ||
+        CHPL_TARGET_PLATFORM == "cray-xc" ||
+        CHPL_TARGET_PLATFORM == "cray-cs") {
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
This feels like a real maintenance burden to me (and one that we will always be reactionary about -- i.e. we will only add new platforms when users file a bug, we will never remember to update this when new platforms are added.)

Surely there has to be a programatic way to check if we're using a posix filesystem

Reply:
(I'd be ok with this going in to fix nightly testing, but at the very least I think we need to open an issue about doing this programmatically)

Reply:
I think it'd be reasonable & acceptable to list the *non-unix* platforms we know about. Is that just cygwin right now?

Reply:
I agree that it is a maintenance burden.  We also have an issue about how to perform different behavior on different file systems (#8146, which I pointed people to earlier), but the answer from the only person who commented on the issue was "punt for now" - performing the work and performing the check in the way you are describing will require figuring this out.

I considered listing all the non-unix platforms, but was concerned about accidentally claiming to support cases that we don't actually support.  If we were able to add support for native Windows as a separate config from cygwin, we would then have isAbsPath approve of a path that is not valid on Windows, silently.  It seems better to give an error when we aren't sure than to fail to give an error when we should.

For logging sake: @dmk42 asked me this question offline and the answer I gave then was that what he was proposed on canonicalizing the filename was another Path function we have proposed but have not yet implemented (but that the developer who contributed this function is working his way towards).

Reply:
>  It seems better to give an error when we aren't sure than to fail to give an error when we should.

Isn't this the role of testing? As long as we have testing that will fail in such a configuration, I'm comfortable with the result. I.e. if we list Cygwin* for now, and somebody adds windows, can we arrange for one of these tests to fail if they ran full local testing?

Reply:
> can we arrange for one of these tests to fail if they ran full local testing?

I don't think we can reasonably predict how to detect an operating system that we haven't supported before.  The windows case was just an example and it happened to be something that exists today.

Reply:
We can reasonably detect when isAbsPath doesn't work on an OS, can't we?
Or is the problem that there might be new & different types of paths that isAbsPath doesn't support? Isn't it sufficient to just say "We assume everything is unix" like we do everywhere else?

Reply:
There's really only posix and windows filesystems (I think this is all boost and python support) -- https://docs.python.org/3/library/os.path.html (python also has an old-style mac path, but that's not really old at this point)

I'm very much in favor of just doing an error for cygwin.

Reply:
Okay, I updated the code and am re-running testing now.  Thanks for the feedback!  Anything else of concern?

Reply:
Upon discussing further with Elliot, he's got a modification to the test file itself so that it will actually detect incorrect usage.